drivers/include/periph/gpio_ll_irq.h

```diff
@@ -0,0 +1,170 @@
+    GPIO_TRIGGER_LEVEL_LOW      /**< level triggered IRQ on low input */
+} gpio_irq_trig_t;
+#endif
+
+/**
+ * @brief   Signature of event callback function that is called on IRQs
+ * @param[in] arg       optional context for the callback
+ * @warning This function is called in IRQ context
+ */
+typedef void (*gpio_ll_cb_t)(void *arg);
+
+/**
+ * @brief   Set up an IRQ for the given GPIO pin and activate it
+ *
+ * @param[in] port      port the pin belongs to
+ * @param[in] pin       number of the pin to setup IRQs on
+ * @param[in] trig      define what triggers the IRQ
+ * @param[in] cb        callback that is called from interrupt context
+ * @param[in] arg       argument to be passed to the callback
+ *
+ * @pre     The given GPIO pin has been initialized using
+ *          @ref gpio_ll_init prior to this call
+ * @pre     The trigger in @p trig is supported by the MCU
+ *
+ * @retval  0           Success
+ * @retval  -ENOTSUP    No external IRQs supported for @p port and @p pin
+ * @retval  -EBUSY      All hardware entities that monitor GPIOs and issue IRQs
+ *                      are busy monitoring other GPIOs. Call
+ *                      @ref gpio_ll_irq_off on one of those and retry.
+ */
+int gpio_ll_irq(gpio_port_t port, uint8_t pin, gpio_irq_trig_t trig,
+                gpio_ll_cb_t cb, void *arg);
+
+/**
+ * @brief   Mask IRQs on the given GPIO pin
+ *
+ * @param[in] port      port the pin belongs to
+ * @param[in] pin       number of the pin to disable IRQs on
+ *
+ * @note    IRQs can be re-enabled without reconfiguration using
+ *          @ref gpio_ll_irq_unmask
+ *
+ * If IRQs are not needed for a longer term, consider using @ref
+ * gpio_ll_irq_off instead.
+ */
+void gpio_ll_irq_mask(gpio_port_t port, uint8_t pin);
+
+#if MODULE_PERIPH_GPIO_LL_IRQ_UNMASK || defined(DOXYGEN)
+/**
+ * @brief   Unmask IRQs on the given GPIO pin
+ *
+ * Same as @ref gpio_ll_irq_unmask_clear except that IRQs that came in during
+ *
+ * @warning On same MCUs (most notably STM32) this is impossible to implement.
+ *          The feature `periph_gpio_ll_irq_unmask` is provided, if this
+ *          function is available.
+ *
+ * @ref gpio_ll_irq_unmask_clear is a portable alternative that might suit your
+ * use case equally well.
+ *
+ * @param[in] port      port the pin belongs to
+ * @param[in] pin       number of the pin to unmask IRQs on
+ */
+void gpio_ll_irq_unmask(gpio_port_t port, uint8_t pin);
+#endif
+
+/**
+ * @brief   Unmask IRQs on the given GPIO pin and clear pending IRQs
+ *
+ * @pre     IRQs on the given pin have been previously configured via
+ *          @ref gpio_ll_irq
+ * @post    IRQs will be acted upon again from now on. If one or more IRQs came
+ *          in while masked, this IRQs are silently lost.
+ *
+ * @param[in] port      port the pin belongs to
+ * @param[in] pin       number of the pin to unmask IRQs on
+ */
+void gpio_ll_irq_unmask_clear(gpio_port_t port, uint8_t pin);
```

Comment:
```suggestion
void gpio_ll_irq_unmask_and_clear(gpio_port_t port, uint8_t pin);
```

We are not so constrained in screen real estate that we need to allow for ambiguity here :wink: